### ISSUE TITLE
Improve lighting

### ISSUE THREAD

**Issue by LaG1924** [LaG1924/AltCraft#19]
At least smooth lighting. Not per-block as now.

Maybe shadows and more realistic light.

**Comment by OneGuyS** [comment]
Maybe shadows and more realistic light.

i would prefer this as optional setting. like rtx on in bedrock. 
also ambient oculusion would be cool

**Comment by LaG1924** [comment]
@OneGuyS 
Yes, this will be optional setting.

Actually, I have already implemented smooth lighting (with some glitches, that I will fix later) and ambient occlusion (SSAO with  deferred rendering #78).

So, I am completely improving overall rendering now, and I want to add realtime directional/point shadows, HDR, maybe PBR.

![image](https://user-images.githubusercontent.com/12997935/160457881-686b4330-20aa-439f-a760-054f812e7ea2.png)

**Comment by OneGuyS** [comment]
So I don't like SDL it's hard to work with. If you want to add prb, advanced lightning or something like this try bgfx. I know it would be hard to port all this work to different library but i thing it would be easier later. 

https://github.com/bkaradzic/bgfx

(I recommend Vulkan RHI.)

**Comment by LaG1924** [comment]
@OneGuyS 
tl;dr: I'm not interested in bgfx.

I know about bgfx and other rendering libraries.
Unfortunately, I already done most 'hard' part of the graphics job. In AltCraft I have my own custom RHI with OpenGL backend (and now I am looking into Vulkan), so I am will implement PBR and other render features using my RHI functions (instead of using OpenGL functions directly), and you suggesting me to replace my RHI with bgfx.
Sorry, but I don't think that I want to use bgfx (maybe later, very later, when there will be no more higher priority tasks), because direct control of graphics API is fun (and can give more performance/features, but main reason is my fun from coding).
SDL used only for input and window management.

**Comment by OneGuyS** [comment]
I'm still impressed by your work and crossing finger for project
